targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_IMX/sleep.c

```diff
@@ -37,8 +36,10 @@ void hal_deepsleep(void)
 {
     vPortPRE_SLEEP_PROCESSING(kCLOCK_ModeStop);
 
-    /* Wait till debug UART is done transmitting */
-    serial_wait_tx_complete(STDIO_UART);
+    /* Check if any of the UART's is transmitting data */
+    if (serial_check_tx_ongoing()) {
+        return;
```

Comment:
Aha! This is actually an approach I hadn't considered, and it is actually much better than waiting or going into shallow sleep. Thank you.

Returning is best here. This gives waking interrupts a chance to operate, rather than sitting there looping with interrupts disabled inside the HAL - possibly bad for performance. The core code will re-enable interrupts briefly, giving them a chance to execute, and then come back here again if it still wants to sleep. The loop will happen at a higher level, with proper "I've changed my mind about sleeping" checks.

Except it looks unbalanced with respect to the `vPortPRE_SLEEP_PROCESSING` above. Should this check be done before that?

Reply:
> Aha! This is actually an approach I hadn't considered, and it is actually much better than waiting or going into shallow sleep. Thank you.
> 
> Returning is best here. This gives waking interrupts a chance to operate, rather than sitting there looping with interrupts disabled inside the HAL - possibly bad for performance. The core code will re-enable interrupts briefly, giving them a chance to execute, and then come back here again if it still wants to sleep. The loop will happen at a higher level, with proper "I've changed my mind about sleeping" checks.
> 
> Except it looks unbalanced with respect to the `vPortPRE_SLEEP_PROCESSING` above. Should this check be done before that?

I have addressed the unbalannce PRE/POST by moving the serial check earlier.